hw/ip/aon_timer/README.md

```diff
@@ -17,52 +20,63 @@ See that document for an overview of how it is integrated into the top level sys
+It has an independent count starting at `wdog_count` and increases on each tick of AON clock.
+The counter can generate interrupts based on the bark and bite thresholds (`wdog_bark_thold` and `wdog_bite_thold`).
 
-When the second threshold is met (this is known as the watchdog bite), a reset request is sent to the power manager which will trigger a system reset.
-This is independent of the IRQ sent as part of the watchdog bark.
-The system reset also resets the always-on timer, so software is not required to directly acknowledge anything after a watchdog reset.
-
-To prevent the watchdog bark or bite, software is expected to periodically reset the count when operating normally.
+To prevent the count reaching the bark or bite thresholds, software is expected to periodically reset the count when operating normally.
 This is referred to as petting the watchdog, and is achieved by resetting the count to zero.
 
-Since this timer functions as a watchdog, it has three additional functions not present in the always-on wakeup timer:
-* Watchdog configuration lock
-* Watchdog pause in sleep
-* Watchdog pause during system escalation
+Since this timer functions as a watchdog, it has two additional features not present in the always-on wakeup timer:
+* Watchdog configuration lock by writing 0 to register `wdog_regwen`
+* Watchdog pause in sleep is enabled if field `wdog_ctrl.pause_in_sleep` is 1 and the counter will halt the count whenever `sleep_mode_i` is set after it's been synchronised.
 
-Unlike the wakeup timer, the watchdog timer configuration can be locked by firmware until the next system reset.
+Once the watchdog timer configuration is locked, firmware cannot modify the timer configuration registers  until the next system reset.
 This allows the option of preventing firmware from accidentally or maliciously disabling the watchdog.
 
-The "pause in sleep" option controls whether the watchdog timer continues to count in low-power modes.
+The "pause in sleep" option controls whether the timer continues to increase in low-power modes if enabled.
 This allows configurations where the watchdog timer can remain programmed and locked while the device is put to sleep for relatively long periods, controlled by the wakeup timer.
 Without this feature, the watchdog timer might wake up the core prematurely by triggering a watchdog bark.
 
-The "pause during escalation" feature ensures that watchdog bites and barks do not interfere with system escalation behavior.
-If during escalation software configures the system to hang instead of reset, the watchdog bite cannot supersede that decision.
+
+#### AON Watchdog bark
+
+Assumming `wdog_ctrl.enable=1` when the count reaches the bark threshold, a level wakeup signal (`wkup_req_o` in AON domain) is sent to the power manager.
+Note `wkup_req_o` can get set by either counter once the threshold is surpassed.
```

Comment:
I'd avoid talking about crossing the threshold and instead make it more static ("if the count is above the bark threshold then..."). I think the result is easier to reason about.